integration/vite-css-build-test.ts

```diff
@@ -9,26 +12,44 @@ import {
 import type { Fixture, AppFixture } from "./helpers/create-fixture.js";
 import { PlaywrightFixture } from "./helpers/playwright-fixture.js";
 
+const __dirname = url.fileURLToPath(new URL(".", import.meta.url));
+
 const TEST_PADDING_VALUE = "20px";
 
 test.describe("Vite CSS build", () => {
   let fixture: Fixture;
   let appFixture: AppFixture;
 
   test.beforeAll(async () => {
+    // set sideEffects for global vanilla extract css
+    let packageJson = JSON.parse(
+      await fs.readFile(
+        path.resolve(__dirname, "helpers", "node-template", "package.json"),
+        "utf-8"
+      )
+    );
+    packageJson.sideEffects = ["*.css.ts"];
+
     fixture = await createFixture({
       compiler: "vite",
       files: {
+        "package.json": JSON.stringify(packageJson, null, 2),
         "remix.config.js": js`
           throw new Error("Remix should not access remix.config.js when using Vite");
           export default {};
         `,
         "vite.config.ts": js`
           import { defineConfig } from "vite";
           import { unstable_vitePlugin as remix } from "@remix-run/dev";
+          import { vanillaExtractPlugin } from "@vanilla-extract/vite-plugin";
 
           export default defineConfig({
-            plugins: [remix()],
+            plugins: [
+              vanillaExtractPlugin({
+                emitCssInSsr: true,
```

Comment:
It surprised me that this is needed. It makes sense that it's needed for our current approach to inlining CSS in dev though, however it's not clear to me why this would be needed for the production build. I'm happy for this PR to go through as-is for now since it's still capturing the current state and proving that it works.

Reply:
Thanks for pointing out.
My intuition was also `emitCssInSsr` should not be necessarily for production build, but I didn't investigate this in detail (actually I haven't even tested turning this off yet...)
I also saw some 

https://github.com/vanilla-extract-css/vanilla-extract/blob/5c9216ca47769d8b042fe46f9662da6a3dc665b1/packages/vite-plugin/src/index.ts#L76-L88

```

Reply:
I'm leaning towards enabling `emitCssInSsr` by default in VE. It feels closer to the way CSS typically works in Vite where it's part of both the server and client graphs. The fact that CSS is missing in the server graph ends up breaking other tools and integrations because this is unexpected. I'm chatting with the rest of the VE team about it.

Adding Remix to this list would definitely be an easy non-breaking fix though.

Reply:
I've opened a PR: https://github.com/vanilla-extract-css/vanilla-extract/pull/1239

Reply:
Thanks for checking in with VE!

I also experimented with toggling `emitCssInSsr` and it seems my intuition was wrong about this.
Probably you know this better, but let me share what I found.

When `emitCssInSsr: false`, the server build `vite build --ssr` will preserve vanilla javascript runtime code like below:

```tsx
// in build/index.js

// originally from app/styles-vanilla-local.css.ts
import { setFileScope, endFileScope } from "@vanilla-extract/css/fileScope";
import { style } from "@vanilla-extract/css";

setFileScope("app/styles-vanilla-local.css.ts", "remix-template-remix");
const index = style({
  background: "lightblue",
  padding: "20px"
});
endFileScope();

// originally from app/routes/_index.tsx
jsx("div", { "data-css-vanilla-local": true, className: index ... })
```

I was thinking `emitCssInSsr: false` means that it still compiles `styles-vanilla-local.css.ts` into hashed class name string (in js) and only skips emitting css code, but that doesn't seem to be the case.

Reply:
That's a fair misunderstanding. If we revisit the options for the plugin, your comment is worth considering.

Reply:
Tests have been updated to use the latest Vanilla Extract Vite plugin where `emitCssInSsr` is no longer needed: https://github.com/remix-run/remix/pull/8063.